static/templates/recipient_row.hbs

```diff
@@ -22,11 +22,7 @@
                 <a class="message_label_clickable narrows_by_topic"
                     href="{{topic_url}}"
                     title="{{#tr this}}Narrow to stream &quot;__display_recipient__&quot;, topic &quot;__topic__&quot;{{/tr}}">
-                    {{#if use_match_properties}}
```

Comment:
Hi @aniketsonu thanks for working on this. This is not the correct approach to solve this. The line was introduced as part of a54704383103812d5d682f06df27f3855f82ec5f which has the history for why it is needed. We need to update the match_topic here after the topic has been updated I think.

Reply:
okok so if we remove this particular snippet of code, the history will not be recorded?

Reply:
Haha! No it is not about keeping history. Removing this line of code will resolve part of the issue but will introduce another issue mentioned which was fixed in the commit id I mentioned.

Reply:
okok, my bad. Will go through the above mentioned commit